ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -0,0 +1,185 @@
+import React, { useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+import { EVENT } from '../../../../shared/constants/metametrics';
+import ErrorMessage from '../../ui/error-message';
+import Popover from '../../ui/popover';
+import Checkbox from '../../ui/check-box';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { I18nContext } from '../../../contexts/i18n';
+import { PageContainerFooter } from '../../ui/page-container';
+import formatMessageParams from './format-message-params';
+import Header from './signature-request-siwe-header';
+import Message from './signature-request-siwe-message';
+
+export default function SignatureRequestSIWE({
+  txData: {
+    msgParams: {
+      version,
+      siwe: { isSIWEDomainValid, isMatchingAddress, messageData },
+    },
+    type,
+  },
+  cancel,
+  sign,
+  fromAccount,
+  subjectMetadata,
+}) {
+  const t = useContext(I18nContext);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const [isShowingDomainWarning, setIsShowingDomainWarning] = useState(false);
+  const [agreeToDomainWarning, setAgreeToDomainWarning] = useState(false);
+
+  const onSign = (event) => {
+    sign(event);
+    trackEvent({
+      category: EVENT.CATEGORIES.TRANSACTIONS,
+      event: 'Confirm',
+      properties: {
+        action: 'SIWE Request',
+        type,
+        version,
+      },
+    });
+  };
+
+  const onCancel = (event) => {
+    cancel(event);
+    trackEvent({
+      category: EVENT.CATEGORIES.TRANSACTIONS,
+      event: 'Cancel',
+      properties: {
+        action: 'SIWE Request',
+        type,
+        version,
+      },
+    });
+  };
+
+  const generateSIWEWarning = () => {
+    let showError = false;
+    let errorMessage = '';
+    let confirmErrorMessage = '';
+
+    if (!isSIWEDomainValid) {
+      showError = true;
+      errorMessage += `${t('SIWEDomainInvalid', [messageData.domain])} `;
+      confirmErrorMessage += `${t('SIWEDomainWarningBody', [
+        messageData.domain,
+      ])} `;
+    }
+
+    if (!isMatchingAddress) {
+      showError = true;
+      errorMessage += `${t('SIWEAddressInvalid', [
+        messageData.address,
+        fromAccount.address,
+      ])} `;
```

Comment:
Double-checking, is this a pattern we have in our application to combine error messages? In this case, we are also mentioning `Proceed with caution` twice to the user. See attachment below
cc: @georgewrmarshall 

<img width="420" alt="Screenshot 2022-05-26 at 5 22 09 PM" src="https://user-images.githubusercontent.com/20778143/170593572-503f2023-b5c7-4115-bc63-e3c5866633ac.png">

Reply:
Great catch! Hmm I have not seen his before. I think it should be probably stack 2 alerts on top of each other. I'm also aware that this might push the buttons further down the page. But I think it's the optimal move here. That error message doesn't read well...

Reply:
Addressed